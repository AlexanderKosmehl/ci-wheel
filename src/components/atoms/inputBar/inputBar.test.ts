import {
  describe, expect, it, jest,
} from '@jest/globals';
import generateInputBar from './inputBar';

describe('generateButton', () => {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const newElementCallback = jest.fn((_newElement: string) => {});

  const inputBar = generateInputBar({
    newElementCallback,
  });
  const inputField = inputBar.querySelector<HTMLInputElement>('div > input');
  const inputButton = inputBar.querySelector<HTMLButtonElement>('div > button');

  it('generates component correctly', () => {
    expect(inputBar).toMatchSnapshot();
  });

  it('adds working callback', () => {
    if (!inputField || !inputButton) throw Error('Missing Inputs!');

    inputField.value = 'Test';
    inputField.dispatchEvent(new Event('keyup'));
    inputButton.click();

    expect(inputField.textContent).toBe('');
    expect(newElementCallback.mock.calls.length).toBe(1);
    expect(newElementCallback.mock.calls[0][0]).toBe('Test');
  });
});
