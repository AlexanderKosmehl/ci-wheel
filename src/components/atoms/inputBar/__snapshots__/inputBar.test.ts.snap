// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateButton generates component correctly 1`] = `
<div
  class="container"
>
  <input
    class="input"
    placeholder="Neuer Eintrag"
    type="text"
  />
  <button
    class="button addButton"
    disabled=""
  >
    <img
      class="icon"
      src="test-file-stub"
    />
  </button>
</div>
`;
